### PR TITLE
DS-508 Implement Popover

### PR DESCRIPTION
**Pull request by jeremy-davis-sonarsource**

### CODE REVIEW

File: src/components/popover/Popover.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ *
+ * ```CSS
+ *   position: relative;
+ *   z-index: 0;
+ * ```
+ *
+ * Since the popovers are appended to the body, they are in the root Stacking Context. If other elements are also there, the z-index will determine which appears on top. By creating a new Stacking Context for your app, it ensures that z-indexed elements will stay within that context, while popovers will be painted on top, in the parent Stacking Context.
+ */
+export const Popover = forwardRef<HTMLButtonElement, Props>((props, ref) => {
+  const { align, children, description, extraContent, footer, title, isOpen, side, ...radixProps } =
```
Comment (david-cho-lerat-sonarsource) [inline src/components/popover/Popover.tsx:74]: OCD NP: isOpen and side are not sorted :grin:

**Review by david-cho-lerat-sonarsource — APPROVED** [review]
Nice! :muscle:

File: src/components/popover/Popover.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+`;
+
+const PopoverFooter = styled.div`
+  margin-top: var(--echoes-dimension-space-200);
+`;
+
+const PopoverContent = styled(RadixPopover.Content)`
+  border: var(--echoes-border-width-default) solid var(--echoes-color-border-weak);
+  border-radius: var(--echoes-border-radius-400);
+  padding: var(--echoes-dimension-space-300) var(--echoes-dimension-space-250);
```
Comment (david-cho-lerat-sonarsource) [inline src/components/popover/Popover.tsx:127]: (this looks huge to me ..)

File: src/components/popover/Popover.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+/* This is the distance between the point of the arrow and the trigger */
+const POPOVER_OFFSET = 4;
+
+/* This is the padding between the edge of the tooltip and the arrow. */
+const ARROW_PADDING = 16;
+
+/**
+ * **Popovers must be attached to a button to be accessible.**
+ *
+ * ### Stacking Context
```
[the remark this answers is not in the document]
Comment (jeremy-davis-sonarsource) [reply src/components/popover/Popover.tsx:60]: Good question! We could add a dedicated page to the migration guide section and point to it from every overlay-type component.

**Pull request merged**